src/codec.ts

```diff
@@ -418,6 +422,10 @@ function encodeValue(value: Value): Value {
     return value.map(encodeValue);
   }
 
+  if (is.object(value)) {
+    return JSON.stringify(value);
+  }
```

Comment:
This is where the most important part of this PR is: It ensures that any value that has been specified as an object will be converted to a valid `JSON` string. An array of objects will be converted to an array of valid `JSON` strings.